// angular2-cookie-law/esm2015/lib/angular2-cookie-law.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * angular2-cookie-law
 *
 * Copyright 2016-2018, @andreasonny83, All rights reserved.
 *
 * @author: @andreasonny83 <andreasonny83@gmail.com>
 */
import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
import { DOCUMENT, isPlatformBrowser } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class Angular2CookieLawService {
    /**
     * @param {?} doc
     * @param {?} platform
     */
    constructor(doc, platform) {
        this.doc = doc;
        this.platform = platform;
    }
    /**
     * @param {?=} cookieName
     * @return {?}
     */
    seen(cookieName = 'cookieLawSeen') {
        let /** @type {?} */ cookies = [];
        if (isPlatformBrowser(this.platform)) {
            cookies = this.doc.cookie.split(';');
        }
        return this.cookieExisits(cookieName, cookies);
    }
    /**
     * @param {?} cookieName
     * @param {?=} expiration
     * @return {?}
     */
    storeCookie(cookieName, expiration) {
        return this.setCookie(cookieName, expiration);
    }
    /**
     * @param {?} name
     * @param {?} cookies
     * @return {?}
     */
    cookieExisits(name, cookies) {
        const /** @type {?} */ cookieName = `${name}=`;
        return cookies.reduce((prev, curr) => prev || curr.trim().search(cookieName) > -1, false);
    }
    /**
     * @param {?} name
     * @param {?=} expiration
     * @return {?}
     */
    setCookie(name, expiration) {
        const /** @type {?} */ now = new Date();
        const /** @type {?} */ exp = new Date(now.getTime() + expiration * 86400000);
        const /** @type {?} */ cookieString = encodeURIComponent(name) + `=true;path=/;expires=${exp.toUTCString()};`;
        if (isPlatformBrowser(this.platform)) {
            this.doc.cookie = cookieString;
        }
    }
}
Angular2CookieLawService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
/** @nocollapse */
Angular2CookieLawService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] },] },
    { type: Object, decorators: [{ type: Inject, args: [PLATFORM_ID,] },] },
];
/** @nocollapse */ Angular2CookieLawService.ngInjectableDef = i0.defineInjectable({ factory: function Angular2CookieLawService_Factory() { return new Angular2CookieLawService(i0.inject(i1.DOCUMENT), i0.inject(i0.PLATFORM_ID)); }, token: Angular2CookieLawService, providedIn: "root" });
function Angular2CookieLawService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    Angular2CookieLawService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    Angular2CookieLawService.ctorParameters;
    /** @type {?} */
    Angular2CookieLawService.prototype.doc;
    /** @type {?} */
    Angular2CookieLawService.prototype.platform;
}
