// angular2-cookie-law/esm5/lib/angular2-cookie-law.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * angular2-cookie-law
 *
 * Copyright 2016-2018, @andreasonny83, All rights reserved.
 *
 * @author: @andreasonny83 <andreasonny83@gmail.com>
 */
import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
import { DOCUMENT, isPlatformBrowser } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
var Angular2CookieLawService = /** @class */ (function () {
    function Angular2CookieLawService(doc, platform) {
        this.doc = doc;
        this.platform = platform;
    }
    /**
     * @param {?=} cookieName
     * @return {?}
     */
    Angular2CookieLawService.prototype.seen = /**
     * @param {?=} cookieName
     * @return {?}
     */
    function (cookieName) {
        if (cookieName === void 0) { cookieName = 'cookieLawSeen'; }
        var /** @type {?} */ cookies = [];
        if (isPlatformBrowser(this.platform)) {
            cookies = this.doc.cookie.split(';');
        }
        return this.cookieExisits(cookieName, cookies);
    };
    /**
     * @param {?} cookieName
     * @param {?=} expiration
     * @return {?}
     */
    Angular2CookieLawService.prototype.storeCookie = /**
     * @param {?} cookieName
     * @param {?=} expiration
     * @return {?}
     */
    function (cookieName, expiration) {
        return this.setCookie(cookieName, expiration);
    };
    /**
     * @param {?} name
     * @param {?} cookies
     * @return {?}
     */
    Angular2CookieLawService.prototype.cookieExisits = /**
     * @param {?} name
     * @param {?} cookies
     * @return {?}
     */
    function (name, cookies) {
        var /** @type {?} */ cookieName = name + "=";
        return cookies.reduce(function (prev, curr) { return prev || curr.trim().search(cookieName) > -1; }, false);
    };
    /**
     * @param {?} name
     * @param {?=} expiration
     * @return {?}
     */
    Angular2CookieLawService.prototype.setCookie = /**
     * @param {?} name
     * @param {?=} expiration
     * @return {?}
     */
    function (name, expiration) {
        var /** @type {?} */ now = new Date();
        var /** @type {?} */ exp = new Date(now.getTime() + expiration * 86400000);
        var /** @type {?} */ cookieString = encodeURIComponent(name) + ("=true;path=/;expires=" + exp.toUTCString() + ";");
        if (isPlatformBrowser(this.platform)) {
            this.doc.cookie = cookieString;
        }
    };
    Angular2CookieLawService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    Angular2CookieLawService.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] },] },
        { type: Object, decorators: [{ type: Inject, args: [PLATFORM_ID,] },] },
    ]; };
    /** @nocollapse */ Angular2CookieLawService.ngInjectableDef = i0.defineInjectable({ factory: function Angular2CookieLawService_Factory() { return new Angular2CookieLawService(i0.inject(i1.DOCUMENT), i0.inject(i0.PLATFORM_ID)); }, token: Angular2CookieLawService, providedIn: "root" });
    return Angular2CookieLawService;
}());
export { Angular2CookieLawService };
function Angular2CookieLawService_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    Angular2CookieLawService.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    Angular2CookieLawService.ctorParameters;
    /** @type {?} */
    Angular2CookieLawService.prototype.doc;
    /** @type {?} */
    Angular2CookieLawService.prototype.platform;
}
