// angular2-cookie-law/esm5/lib/angular2-cookie-law.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * angular2-cookie-law
 *
 * Copyright 2016-2018, @andreasonny83, All rights reserved.
 *
 * @author: @andreasonny83 <andreasonny83@gmail.com>
 */
import { Component, HostBinding, ViewEncapsulation, Input, Output, EventEmitter } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import { closeIcon } from './icons';
import { translateInOut } from './animations';
var CookieLawComponent = /** @class */ (function () {
    function CookieLawComponent(domSanitizer) {
        this.domSanitizer = domSanitizer;
        this.isSeen = new EventEmitter();
        this.transition = 'bottomIn';
        this._position = 'bottom';
        this.cookieLawClass = true;
    }
    Object.defineProperty(CookieLawComponent.prototype, "learnMore", {
        get: /**
         * @return {?}
         */
        function () {
            return this._learnMore;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this._learnMore = value !== null && "" + value !== 'false' ? value : null;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CookieLawComponent.prototype, "awsomeCloseIcon", {
        get: /**
         * @return {?}
         */
        function () {
            return this._awsomeCloseIcon;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this._awsomeCloseIcon =
                value !== null && "" + value !== 'false' ? value : null;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CookieLawComponent.prototype, "target", {
        get: /**
         * @return {?}
         */
        function () {
            return this._target;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this._target =
                value !== null &&
                    "" + value !== 'false' &&
                    ("" + value === '_blank' || "" + value === '_self')
                    ? value
                    : '_blank';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CookieLawComponent.prototype, "position", {
        get: /**
         * @return {?}
         */
        function () {
            return this._position;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this._position =
                value !== null &&
                    "" + value !== 'false' &&
                    ("" + value === 'top' || "" + value === 'bottom')
                    ? value
                    : 'bottom';
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    CookieLawComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.noopener = this._target === '_blank';
        this.transition = this.position === 'bottom' ? 'bottomIn' : 'topIn';
        if (this._awsomeCloseIcon) {
            this.closeSvg = this.domSanitizer.bypassSecurityTrustHtml("<i class=\"fab " + this._awsomeCloseIcon + "\"></i>");
        }
        else {
            this.closeSvg = this.domSanitizer.bypassSecurityTrustHtml(closeIcon);
        }
        this.currentStyles = {
            top: this.position === 'top' ? '0' : null,
            bottom: this.position === 'top' ? 'initial' : null
        };
    };
    /**
     * @param {?} evt
     * @return {?}
     */
    CookieLawComponent.prototype.afterDismissAnimation = /**
     * @param {?} evt
     * @return {?}
     */
    function (evt) {
        if (evt.toState === 'topOut' || evt.toState === 'bottomOut') {
            this.isSeen.emit(true);
        }
    };
    /**
     * @param {?=} evt
     * @return {?}
     */
    CookieLawComponent.prototype.dismiss = /**
     * @param {?=} evt
     * @return {?}
     */
    function (evt) {
        if (evt) {
            evt.preventDefault();
        }
        this.transition = this.position === 'top' ? 'topOut' : 'bottomOut';
    };
    CookieLawComponent.decorators = [
        { type: Component, args: [{
                    selector: 'cookie-law-component',
                    template: "<div\n  class=\"cookie-law-wrapper\"\n  [ngStyle]=\"currentStyles\"\n  [@transition]=\"transition\"\n  (@transition.done)=\"afterDismissAnimation($event)\"\n>\n  <div class=\"copy\">\n    <span #ref><ng-content></ng-content></span>\n\n    <span *ngIf=\"ref.childNodes.length == 0\">\n      By continuing to browse the site, you're agreeing to our use of cookies.\n      <span *ngIf=\"learnMore\">\n        Learn more in our\n        <a [href]=\"learnMore\" [target]=\"target\" [rel]=\"noopener ? 'noopener noreferrer' : null\"\n          >privacy policy</a\n        >.\n      </span>\n    </span>\n\n    <a\n      href=\"#\"\n      role=\"button\"\n      class=\"dismiss\"\n      [innerHTML]=\"closeSvg\"\n      (click)=\"dismiss($event)\"\n      [attr.aria-label]=\"closeSvg\"\n    ></a>\n  </div>\n</div>\n",
                    styles: [".cookie-law-wrapper{background:#333;color:#bbb;display:block;font-family:Helvetica Neue,Helvetica,Arial,sans-serif;font-size:15px;font-weight:200;line-height:20px;position:fixed;bottom:0;left:0;width:100%;z-index:999999999;font-smooth:always;-webkit-font-smoothing:antialiased}.cookie-law-wrapper a{color:#bbb;transition:color .2s}.cookie-law-wrapper a:hover{color:#fff}.cookie-law-wrapper a:hover svg{fill:#fff}.cookie-law-wrapper .dismiss{display:block;box-sizing:border-box;padding:10px;position:absolute;top:0;right:10px;text-decoration:none;line-height:20px}.cookie-law-wrapper .dismiss svg{display:block;fill:#bbb;width:20px;height:20px;transition:fill .2s}.cookie-law-wrapper .copy{box-sizing:border-box;padding:10px 60px 10px 10px}.cookie-law-wrapper .copy span{color:#fff;font-weight:400}.cookie-law-wrapper .copy a{text-decoration:underline}.cookie-law-wrapper .copy a:active,.copy a:hover{outline:0}@media (min-width:600px){.cookie-law-wrapper .copy{padding:20px 60px 20px 20px;font-size:18px;line-height:24px}.cookie-law-wrapper .dismiss{top:10px;right:15px}.cookie-law-wrapper .dismiss svg{width:24px;height:24px}}"],
                    animations: [translateInOut],
                    encapsulation: ViewEncapsulation.None
                },] },
    ];
    /** @nocollapse */
    CookieLawComponent.ctorParameters = function () { return [
        { type: DomSanitizer, },
    ]; };
    CookieLawComponent.propDecorators = {
        "cookieLawClass": [{ type: HostBinding, args: ['class.cookie-law',] },],
        "learnMore": [{ type: Input },],
        "awsomeCloseIcon": [{ type: Input },],
        "target": [{ type: Input },],
        "position": [{ type: Input },],
        "isSeen": [{ type: Output },],
    };
    return CookieLawComponent;
}());
export { CookieLawComponent };
function CookieLawComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    CookieLawComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    CookieLawComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    CookieLawComponent.propDecorators;
    /** @type {?} */
    CookieLawComponent.prototype.closeSvg;
    /** @type {?} */
    CookieLawComponent.prototype.currentStyles;
    /** @type {?} */
    CookieLawComponent.prototype.transition;
    /** @type {?} */
    CookieLawComponent.prototype.cookieLawClass;
    /** @type {?} */
    CookieLawComponent.prototype.isSeen;
    /** @type {?} */
    CookieLawComponent.prototype.noopener;
    /** @type {?} */
    CookieLawComponent.prototype._learnMore;
    /** @type {?} */
    CookieLawComponent.prototype._awsomeCloseIcon;
    /** @type {?} */
    CookieLawComponent.prototype._target;
    /** @type {?} */
    CookieLawComponent.prototype._position;
    /** @type {?} */
    CookieLawComponent.prototype.domSanitizer;
}
