// angular2-cookie-law/esm5/lib/angular2-cookie-law-container.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * angular2-cookie-law
 *
 * Copyright 2016-2018, @andreasonny83, All rights reserved.
 *
 * @author: @andreasonny83 <andreasonny83@gmail.com>
 */
import { Component, ViewChild, HostBinding, Input, Output, EventEmitter } from '@angular/core';
import { Angular2CookieLawService } from './angular2-cookie-law.service';
import { CookieLawComponent } from './angular2-cookie-law.component';
var CookieLawContainerComponent = /** @class */ (function () {
    function CookieLawContainerComponent(cookieLawService) {
        this.cookieLawService = cookieLawService;
        this.isSeen = new EventEmitter();
        this.name = 'cookieLawSeen'; // set a default cookie name if not provided
        this.seen = true;
    }
    Object.defineProperty(CookieLawContainerComponent.prototype, "cookieLawSeen", {
        get: /**
         * @return {?}
         */
        function () {
            return this.cookieLawService.seen(this.name);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    CookieLawContainerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.seen = this.cookieLawService.seen(this.name);
    };
    /**
     * @return {?}
     */
    CookieLawContainerComponent.prototype.hasBeenDismissed = /**
     * @return {?}
     */
    function () {
        this.cookieLawService.storeCookie(this.name, this.expiration);
        this.seen = true;
        this.isSeen.emit(true);
    };
    /**
     * @return {?}
     */
    CookieLawContainerComponent.prototype.dismiss = /**
     * @return {?}
     */
    function () {
        this.cookieLawComponent.dismiss();
    };
    CookieLawContainerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'cookie-law',
                    template: "\n    <cookie-law-component\n      *ngIf=\"!seen\"\n      [awsomeCloseIcon]=\"awsomeCloseIcon\"\n      [learnMore]=\"learnMore\"\n      [target]=\"target\"\n      [position]=\"position\"\n      (isSeen)=\"hasBeenDismissed()\"\n    >\n      <ng-content></ng-content>\n    </cookie-law-component>\n  "
                },] },
    ];
    /** @nocollapse */
    CookieLawContainerComponent.ctorParameters = function () { return [
        { type: Angular2CookieLawService, },
    ]; };
    CookieLawContainerComponent.propDecorators = {
        "seen": [{ type: HostBinding, args: ['attr.seen',] },],
        "cookieLawComponent": [{ type: ViewChild, args: [CookieLawComponent,] },],
        "name": [{ type: Input },],
        "learnMore": [{ type: Input },],
        "target": [{ type: Input },],
        "position": [{ type: Input },],
        "expiration": [{ type: Input },],
        "awsomeCloseIcon": [{ type: Input },],
        "isSeen": [{ type: Output },],
    };
    return CookieLawContainerComponent;
}());
export { CookieLawContainerComponent };
function CookieLawContainerComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    CookieLawContainerComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    CookieLawContainerComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    CookieLawContainerComponent.propDecorators;
    /** @type {?} */
    CookieLawContainerComponent.prototype.seen;
    /** @type {?} */
    CookieLawContainerComponent.prototype.cookieLawComponent;
    /** @type {?} */
    CookieLawContainerComponent.prototype.name;
    /** @type {?} */
    CookieLawContainerComponent.prototype.learnMore;
    /** @type {?} */
    CookieLawContainerComponent.prototype.target;
    /** @type {?} */
    CookieLawContainerComponent.prototype.position;
    /** @type {?} */
    CookieLawContainerComponent.prototype.expiration;
    /** @type {?} */
    CookieLawContainerComponent.prototype.awsomeCloseIcon;
    /** @type {?} */
    CookieLawContainerComponent.prototype.isSeen;
    /** @type {?} */
    CookieLawContainerComponent.prototype.cookieLawService;
}
