// angular2-cookie-law/fesm5/angular2-cookie-law.js
import { Inject, Injectable, PLATFORM_ID, Component, ViewChild, HostBinding, Input, Output, EventEmitter, ViewEncapsulation, NgModule, Optional, SkipSelf, defineInjectable, inject } from '@angular/core';
import { DOCUMENT, isPlatformBrowser, CommonModule } from '@angular/common';
import { trigger, state, style, animate, transition } from '@angular/animations';
import { DomSanitizer } from '@angular/platform-browser';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var Angular2CookieLawService = /** @class */ (function () {
    function Angular2CookieLawService(doc, platform) {
        this.doc = doc;
        this.platform = platform;
    }
    /**
     * @param {?=} cookieName
     * @return {?}
     */
    Angular2CookieLawService.prototype.seen = /**
     * @param {?=} cookieName
     * @return {?}
     */
    function (cookieName) {
        if (cookieName === void 0) { cookieName = 'cookieLawSeen'; }
        var /** @type {?} */ cookies = [];
        if (isPlatformBrowser(this.platform)) {
            cookies = this.doc.cookie.split(';');
        }
        return this.cookieExisits(cookieName, cookies);
    };
    /**
     * @param {?} cookieName
     * @param {?=} expiration
     * @return {?}
     */
    Angular2CookieLawService.prototype.storeCookie = /**
     * @param {?} cookieName
     * @param {?=} expiration
     * @return {?}
     */
    function (cookieName, expiration) {
        return this.setCookie(cookieName, expiration);
    };
    /**
     * @param {?} name
     * @param {?} cookies
     * @return {?}
     */
    Angular2CookieLawService.prototype.cookieExisits = /**
     * @param {?} name
     * @param {?} cookies
     * @return {?}
     */
    function (name, cookies) {
        var /** @type {?} */ cookieName = name + "=";
        return cookies.reduce(function (prev, curr) { return prev || curr.trim().search(cookieName) > -1; }, false);
    };
    /**
     * @param {?} name
     * @param {?=} expiration
     * @return {?}
     */
    Angular2CookieLawService.prototype.setCookie = /**
     * @param {?} name
     * @param {?=} expiration
     * @return {?}
     */
    function (name, expiration) {
        var /** @type {?} */ now = new Date();
        var /** @type {?} */ exp = new Date(now.getTime() + expiration * 86400000);
        var /** @type {?} */ cookieString = encodeURIComponent(name) + ("=true;path=/;expires=" + exp.toUTCString() + ";");
        if (isPlatformBrowser(this.platform)) {
            this.doc.cookie = cookieString;
        }
    };
    Angular2CookieLawService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    Angular2CookieLawService.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] },] },
        { type: Object, decorators: [{ type: Inject, args: [PLATFORM_ID,] },] },
    ]; };
    /** @nocollapse */ Angular2CookieLawService.ngInjectableDef = defineInjectable({ factory: function Angular2CookieLawService_Factory() { return new Angular2CookieLawService(inject(DOCUMENT), inject(PLATFORM_ID)); }, token: Angular2CookieLawService, providedIn: "root" });
    return Angular2CookieLawService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * angular2-cookie-law
 *
 * Copyright 2016-2018, @andreasonny83, All rights reserved.
 *
 * @author: @andreasonny83 <andreasonny83@gmail.com>
 */
var /** @type {?} */ closeIcon = "\n  <svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 512 512\">\n    <path d=\"M377.047 184.198q0 8.26-6.037 14.297L313.505 256l57.505 57.505q6.037 6.037 6.037 14.297 0 8.578-6.037\n      14.615l-28.593 28.593q-6.037 6.037-14.615 6.037-8.26 0-14.297-6.037L256 313.505l-57.505 57.505q-6.037\n      6.037-14.297 6.037-8.578 0-14.615-6.037l-28.593-28.593q-6.037-6.037-6.037-14.615 0-8.26 6.037-14.297L198.495\n      256l-57.505-57.505q-6.037-6.037-6.037-14.297 0-8.578 6.037-14.615l28.593-28.593q6.037-6.037 14.615-6.037 8.26 0\n      14.297 6.037L256 198.495l57.505-57.505q6.037-6.037 14.297-6.037 8.578 0 14.615 6.037l28.593 28.593q6.037 6.037\n      6.037 14.615zM500 256q0-66.401-32.724-122.477-32.724-56.075-88.799-88.799Q322.401 12 256 12q-66.401 0-122.477\n      32.724-56.075 32.724-88.799 88.799Q12 189.599 12 256q0 66.401 32.724 122.477 32.724 56.075 88.799 88.799Q189.599\n      500 256 500q66.401 0 122.477-32.724 56.075-32.724 88.799-88.799Q500 322.401 500 256z\"></path>\n  </svg>";

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var /** @type {?} */ translateInOut = trigger('transition', [
    state('*', style({ transform: 'translateY(0)' })),
    state('void', style({ transform: 'translateY(0)' })),
    state('bottomOut', style({ transform: 'translateY(100%)' })),
    state('topOut', style({ transform: 'translateY(-100%)' })),
    transition('void => topIn', [
        style({ transform: 'translateY(-100%)' }),
        animate('1000ms ease-in-out')
    ]),
    transition('void => bottomIn', [
        style({ transform: 'translateY(100%)' }),
        animate('1000ms ease-in-out')
    ]),
    transition('* => *', animate('1000ms ease-out'))
]);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var CookieLawComponent = /** @class */ (function () {
    function CookieLawComponent(domSanitizer) {
        this.domSanitizer = domSanitizer;
        this.isSeen = new EventEmitter();
        this.transition = 'bottomIn';
        this._position = 'bottom';
        this.cookieLawClass = true;
    }
    Object.defineProperty(CookieLawComponent.prototype, "learnMore", {
        get: /**
         * @return {?}
         */
        function () {
            return this._learnMore;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this._learnMore = value !== null && "" + value !== 'false' ? value : null;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CookieLawComponent.prototype, "awsomeCloseIcon", {
        get: /**
         * @return {?}
         */
        function () {
            return this._awsomeCloseIcon;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this._awsomeCloseIcon =
                value !== null && "" + value !== 'false' ? value : null;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CookieLawComponent.prototype, "target", {
        get: /**
         * @return {?}
         */
        function () {
            return this._target;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this._target =
                value !== null &&
                    "" + value !== 'false' &&
                    ("" + value === '_blank' || "" + value === '_self')
                    ? value
                    : '_blank';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(CookieLawComponent.prototype, "position", {
        get: /**
         * @return {?}
         */
        function () {
            return this._position;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this._position =
                value !== null &&
                    "" + value !== 'false' &&
                    ("" + value === 'top' || "" + value === 'bottom')
                    ? value
                    : 'bottom';
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    CookieLawComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.noopener = this._target === '_blank';
        this.transition = this.position === 'bottom' ? 'bottomIn' : 'topIn';
        if (this._awsomeCloseIcon) {
            this.closeSvg = this.domSanitizer.bypassSecurityTrustHtml("<i class=\"fab " + this._awsomeCloseIcon + "\"></i>");
        }
        else {
            this.closeSvg = this.domSanitizer.bypassSecurityTrustHtml(closeIcon);
        }
        this.currentStyles = {
            top: this.position === 'top' ? '0' : null,
            bottom: this.position === 'top' ? 'initial' : null
        };
    };
    /**
     * @param {?} evt
     * @return {?}
     */
    CookieLawComponent.prototype.afterDismissAnimation = /**
     * @param {?} evt
     * @return {?}
     */
    function (evt) {
        if (evt.toState === 'topOut' || evt.toState === 'bottomOut') {
            this.isSeen.emit(true);
        }
    };
    /**
     * @param {?=} evt
     * @return {?}
     */
    CookieLawComponent.prototype.dismiss = /**
     * @param {?=} evt
     * @return {?}
     */
    function (evt) {
        if (evt) {
            evt.preventDefault();
        }
        this.transition = this.position === 'top' ? 'topOut' : 'bottomOut';
    };
    CookieLawComponent.decorators = [
        { type: Component, args: [{
                    selector: 'cookie-law-component',
                    template: "<div\n  class=\"cookie-law-wrapper\"\n  [ngStyle]=\"currentStyles\"\n  [@transition]=\"transition\"\n  (@transition.done)=\"afterDismissAnimation($event)\"\n>\n  <div class=\"copy\">\n    <span #ref><ng-content></ng-content></span>\n\n    <span *ngIf=\"ref.childNodes.length == 0\">\n      By continuing to browse the site, you're agreeing to our use of cookies.\n      <span *ngIf=\"learnMore\">\n        Learn more in our\n        <a [href]=\"learnMore\" [target]=\"target\" [rel]=\"noopener ? 'noopener noreferrer' : null\"\n          >privacy policy</a\n        >.\n      </span>\n    </span>\n\n    <a\n      href=\"#\"\n      role=\"button\"\n      class=\"dismiss\"\n      [innerHTML]=\"closeSvg\"\n      (click)=\"dismiss($event)\"\n      [attr.aria-label]=\"closeSvg\"\n    ></a>\n  </div>\n</div>\n",
                    styles: [".cookie-law-wrapper{background:#333;color:#bbb;display:block;font-family:Helvetica Neue,Helvetica,Arial,sans-serif;font-size:15px;font-weight:200;line-height:20px;position:fixed;bottom:0;left:0;width:100%;z-index:999999999;font-smooth:always;-webkit-font-smoothing:antialiased}.cookie-law-wrapper a{color:#bbb;transition:color .2s}.cookie-law-wrapper a:hover{color:#fff}.cookie-law-wrapper a:hover svg{fill:#fff}.cookie-law-wrapper .dismiss{display:block;box-sizing:border-box;padding:10px;position:absolute;top:0;right:10px;text-decoration:none;line-height:20px}.cookie-law-wrapper .dismiss svg{display:block;fill:#bbb;width:20px;height:20px;transition:fill .2s}.cookie-law-wrapper .copy{box-sizing:border-box;padding:10px 60px 10px 10px}.cookie-law-wrapper .copy span{color:#fff;font-weight:400}.cookie-law-wrapper .copy a{text-decoration:underline}.cookie-law-wrapper .copy a:active,.copy a:hover{outline:0}@media (min-width:600px){.cookie-law-wrapper .copy{padding:20px 60px 20px 20px;font-size:18px;line-height:24px}.cookie-law-wrapper .dismiss{top:10px;right:15px}.cookie-law-wrapper .dismiss svg{width:24px;height:24px}}"],
                    animations: [translateInOut],
                    encapsulation: ViewEncapsulation.None
                },] },
    ];
    /** @nocollapse */
    CookieLawComponent.ctorParameters = function () { return [
        { type: DomSanitizer, },
    ]; };
    CookieLawComponent.propDecorators = {
        "cookieLawClass": [{ type: HostBinding, args: ['class.cookie-law',] },],
        "learnMore": [{ type: Input },],
        "awsomeCloseIcon": [{ type: Input },],
        "target": [{ type: Input },],
        "position": [{ type: Input },],
        "isSeen": [{ type: Output },],
    };
    return CookieLawComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var CookieLawContainerComponent = /** @class */ (function () {
    function CookieLawContainerComponent(cookieLawService) {
        this.cookieLawService = cookieLawService;
        this.isSeen = new EventEmitter();
        this.name = 'cookieLawSeen'; // set a default cookie name if not provided
        this.seen = true;
    }
    Object.defineProperty(CookieLawContainerComponent.prototype, "cookieLawSeen", {
        get: /**
         * @return {?}
         */
        function () {
            return this.cookieLawService.seen(this.name);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    CookieLawContainerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.seen = this.cookieLawService.seen(this.name);
    };
    /**
     * @return {?}
     */
    CookieLawContainerComponent.prototype.hasBeenDismissed = /**
     * @return {?}
     */
    function () {
        this.cookieLawService.storeCookie(this.name, this.expiration);
        this.seen = true;
        this.isSeen.emit(true);
    };
    /**
     * @return {?}
     */
    CookieLawContainerComponent.prototype.dismiss = /**
     * @return {?}
     */
    function () {
        this.cookieLawComponent.dismiss();
    };
    CookieLawContainerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'cookie-law',
                    template: "\n    <cookie-law-component\n      *ngIf=\"!seen\"\n      [awsomeCloseIcon]=\"awsomeCloseIcon\"\n      [learnMore]=\"learnMore\"\n      [target]=\"target\"\n      [position]=\"position\"\n      (isSeen)=\"hasBeenDismissed()\"\n    >\n      <ng-content></ng-content>\n    </cookie-law-component>\n  "
                },] },
    ];
    /** @nocollapse */
    CookieLawContainerComponent.ctorParameters = function () { return [
        { type: Angular2CookieLawService, },
    ]; };
    CookieLawContainerComponent.propDecorators = {
        "seen": [{ type: HostBinding, args: ['attr.seen',] },],
        "cookieLawComponent": [{ type: ViewChild, args: [CookieLawComponent,] },],
        "name": [{ type: Input },],
        "learnMore": [{ type: Input },],
        "target": [{ type: Input },],
        "position": [{ type: Input },],
        "expiration": [{ type: Input },],
        "awsomeCloseIcon": [{ type: Input },],
        "isSeen": [{ type: Output },],
    };
    return CookieLawContainerComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var CookieLawModule = /** @class */ (function () {
    function CookieLawModule(parentModule) {
        if (parentModule) {
            throw new Error('CookieLawModule is already loaded. Import it in the AppModule only');
        }
    }
    CookieLawModule.decorators = [
        { type: NgModule, args: [{
                    imports: [CommonModule],
                    declarations: [CookieLawComponent, CookieLawContainerComponent],
                    exports: [CookieLawContainerComponent]
                },] },
    ];
    /** @nocollapse */
    CookieLawModule.ctorParameters = function () { return [
        { type: CookieLawModule, decorators: [{ type: Optional }, { type: SkipSelf },] },
    ]; };
    return CookieLawModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { Angular2CookieLawService, CookieLawContainerComponent, CookieLawComponent, CookieLawModule, translateInOut as ɵa };
