// angular2-cookie-law/fesm2015/angular2-cookie-law.js
import { Inject, Injectable, PLATFORM_ID, Component, ViewChild, HostBinding, Input, Output, EventEmitter, ViewEncapsulation, NgModule, Optional, SkipSelf, defineInjectable, inject } from '@angular/core';
import { DOCUMENT, isPlatformBrowser, CommonModule } from '@angular/common';
import { trigger, state, style, animate, transition } from '@angular/animations';
import { DomSanitizer } from '@angular/platform-browser';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class Angular2CookieLawService {
    /**
     * @param {?} doc
     * @param {?} platform
     */
    constructor(doc, platform) {
        this.doc = doc;
        this.platform = platform;
    }
    /**
     * @param {?=} cookieName
     * @return {?}
     */
    seen(cookieName = 'cookieLawSeen') {
        let /** @type {?} */ cookies = [];
        if (isPlatformBrowser(this.platform)) {
            cookies = this.doc.cookie.split(';');
        }
        return this.cookieExisits(cookieName, cookies);
    }
    /**
     * @param {?} cookieName
     * @param {?=} expiration
     * @return {?}
     */
    storeCookie(cookieName, expiration) {
        return this.setCookie(cookieName, expiration);
    }
    /**
     * @param {?} name
     * @param {?} cookies
     * @return {?}
     */
    cookieExisits(name, cookies) {
        const /** @type {?} */ cookieName = `${name}=`;
        return cookies.reduce((prev, curr) => prev || curr.trim().search(cookieName) > -1, false);
    }
    /**
     * @param {?} name
     * @param {?=} expiration
     * @return {?}
     */
    setCookie(name, expiration) {
        const /** @type {?} */ now = new Date();
        const /** @type {?} */ exp = new Date(now.getTime() + expiration * 86400000);
        const /** @type {?} */ cookieString = encodeURIComponent(name) + `=true;path=/;expires=${exp.toUTCString()};`;
        if (isPlatformBrowser(this.platform)) {
            this.doc.cookie = cookieString;
        }
    }
}
Angular2CookieLawService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
/** @nocollapse */
Angular2CookieLawService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] },] },
    { type: Object, decorators: [{ type: Inject, args: [PLATFORM_ID,] },] },
];
/** @nocollapse */ Angular2CookieLawService.ngInjectableDef = defineInjectable({ factory: function Angular2CookieLawService_Factory() { return new Angular2CookieLawService(inject(DOCUMENT), inject(PLATFORM_ID)); }, token: Angular2CookieLawService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * angular2-cookie-law
 *
 * Copyright 2016-2018, @andreasonny83, All rights reserved.
 *
 * @author: @andreasonny83 <andreasonny83@gmail.com>
 */
const /** @type {?} */ closeIcon = `
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
    <path d="M377.047 184.198q0 8.26-6.037 14.297L313.505 256l57.505 57.505q6.037 6.037 6.037 14.297 0 8.578-6.037
      14.615l-28.593 28.593q-6.037 6.037-14.615 6.037-8.26 0-14.297-6.037L256 313.505l-57.505 57.505q-6.037
      6.037-14.297 6.037-8.578 0-14.615-6.037l-28.593-28.593q-6.037-6.037-6.037-14.615 0-8.26 6.037-14.297L198.495
      256l-57.505-57.505q-6.037-6.037-6.037-14.297 0-8.578 6.037-14.615l28.593-28.593q6.037-6.037 14.615-6.037 8.26 0
      14.297 6.037L256 198.495l57.505-57.505q6.037-6.037 14.297-6.037 8.578 0 14.615 6.037l28.593 28.593q6.037 6.037
      6.037 14.615zM500 256q0-66.401-32.724-122.477-32.724-56.075-88.799-88.799Q322.401 12 256 12q-66.401 0-122.477
      32.724-56.075 32.724-88.799 88.799Q12 189.599 12 256q0 66.401 32.724 122.477 32.724 56.075 88.799 88.799Q189.599
      500 256 500q66.401 0 122.477-32.724 56.075-32.724 88.799-88.799Q500 322.401 500 256z"></path>
  </svg>`;

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
const /** @type {?} */ translateInOut = trigger('transition', [
    state('*', style({ transform: 'translateY(0)' })),
    state('void', style({ transform: 'translateY(0)' })),
    state('bottomOut', style({ transform: 'translateY(100%)' })),
    state('topOut', style({ transform: 'translateY(-100%)' })),
    transition('void => topIn', [
        style({ transform: 'translateY(-100%)' }),
        animate('1000ms ease-in-out')
    ]),
    transition('void => bottomIn', [
        style({ transform: 'translateY(100%)' }),
        animate('1000ms ease-in-out')
    ]),
    transition('* => *', animate('1000ms ease-out'))
]);

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class CookieLawComponent {
    /**
     * @param {?} domSanitizer
     */
    constructor(domSanitizer) {
        this.domSanitizer = domSanitizer;
        this.isSeen = new EventEmitter();
        this.transition = 'bottomIn';
        this._position = 'bottom';
        this.cookieLawClass = true;
    }
    /**
     * @return {?}
     */
    get learnMore() {
        return this._learnMore;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set learnMore(value) {
        this._learnMore = value !== null && `${value}` !== 'false' ? value : null;
    }
    /**
     * @return {?}
     */
    get awsomeCloseIcon() {
        return this._awsomeCloseIcon;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set awsomeCloseIcon(value) {
        this._awsomeCloseIcon =
            value !== null && `${value}` !== 'false' ? value : null;
    }
    /**
     * @return {?}
     */
    get target() {
        return this._target;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set target(value) {
        this._target =
            value !== null &&
                `${value}` !== 'false' &&
                (`${value}` === '_blank' || `${value}` === '_self')
                ? value
                : '_blank';
    }
    /**
     * @return {?}
     */
    get position() {
        return this._position;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set position(value) {
        this._position =
            value !== null &&
                `${value}` !== 'false' &&
                (`${value}` === 'top' || `${value}` === 'bottom')
                ? value
                : 'bottom';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.noopener = this._target === '_blank';
        this.transition = this.position === 'bottom' ? 'bottomIn' : 'topIn';
        if (this._awsomeCloseIcon) {
            this.closeSvg = this.domSanitizer.bypassSecurityTrustHtml(`<i class="fab ${this._awsomeCloseIcon}"></i>`);
        }
        else {
            this.closeSvg = this.domSanitizer.bypassSecurityTrustHtml(closeIcon);
        }
        this.currentStyles = {
            top: this.position === 'top' ? '0' : null,
            bottom: this.position === 'top' ? 'initial' : null
        };
    }
    /**
     * @param {?} evt
     * @return {?}
     */
    afterDismissAnimation(evt) {
        if (evt.toState === 'topOut' || evt.toState === 'bottomOut') {
            this.isSeen.emit(true);
        }
    }
    /**
     * @param {?=} evt
     * @return {?}
     */
    dismiss(evt) {
        if (evt) {
            evt.preventDefault();
        }
        this.transition = this.position === 'top' ? 'topOut' : 'bottomOut';
    }
}
CookieLawComponent.decorators = [
    { type: Component, args: [{
                selector: 'cookie-law-component',
                template: `<div
  class="cookie-law-wrapper"
  [ngStyle]="currentStyles"
  [@transition]="transition"
  (@transition.done)="afterDismissAnimation($event)"
>
  <div class="copy">
    <span #ref><ng-content></ng-content></span>

    <span *ngIf="ref.childNodes.length == 0">
      By continuing to browse the site, you're agreeing to our use of cookies.
      <span *ngIf="learnMore">
        Learn more in our
        <a [href]="learnMore" [target]="target" [rel]="noopener ? 'noopener noreferrer' : null"
          >privacy policy</a
        >.
      </span>
    </span>

    <a
      href="#"
      role="button"
      class="dismiss"
      [innerHTML]="closeSvg"
      (click)="dismiss($event)"
      [attr.aria-label]="closeSvg"
    ></a>
  </div>
</div>
`,
                styles: [`.cookie-law-wrapper{background:#333;color:#bbb;display:block;font-family:Helvetica Neue,Helvetica,Arial,sans-serif;font-size:15px;font-weight:200;line-height:20px;position:fixed;bottom:0;left:0;width:100%;z-index:999999999;font-smooth:always;-webkit-font-smoothing:antialiased}.cookie-law-wrapper a{color:#bbb;transition:color .2s}.cookie-law-wrapper a:hover{color:#fff}.cookie-law-wrapper a:hover svg{fill:#fff}.cookie-law-wrapper .dismiss{display:block;box-sizing:border-box;padding:10px;position:absolute;top:0;right:10px;text-decoration:none;line-height:20px}.cookie-law-wrapper .dismiss svg{display:block;fill:#bbb;width:20px;height:20px;transition:fill .2s}.cookie-law-wrapper .copy{box-sizing:border-box;padding:10px 60px 10px 10px}.cookie-law-wrapper .copy span{color:#fff;font-weight:400}.cookie-law-wrapper .copy a{text-decoration:underline}.cookie-law-wrapper .copy a:active,.copy a:hover{outline:0}@media (min-width:600px){.cookie-law-wrapper .copy{padding:20px 60px 20px 20px;font-size:18px;line-height:24px}.cookie-law-wrapper .dismiss{top:10px;right:15px}.cookie-law-wrapper .dismiss svg{width:24px;height:24px}}`],
                animations: [translateInOut],
                encapsulation: ViewEncapsulation.None
            },] },
];
/** @nocollapse */
CookieLawComponent.ctorParameters = () => [
    { type: DomSanitizer, },
];
CookieLawComponent.propDecorators = {
    "cookieLawClass": [{ type: HostBinding, args: ['class.cookie-law',] },],
    "learnMore": [{ type: Input },],
    "awsomeCloseIcon": [{ type: Input },],
    "target": [{ type: Input },],
    "position": [{ type: Input },],
    "isSeen": [{ type: Output },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class CookieLawContainerComponent {
    /**
     * @param {?} cookieLawService
     */
    constructor(cookieLawService) {
        this.cookieLawService = cookieLawService;
        this.isSeen = new EventEmitter();
        this.name = 'cookieLawSeen'; // set a default cookie name if not provided
        this.seen = true;
    }
    /**
     * @return {?}
     */
    get cookieLawSeen() {
        return this.cookieLawService.seen(this.name);
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.seen = this.cookieLawService.seen(this.name);
    }
    /**
     * @return {?}
     */
    hasBeenDismissed() {
        this.cookieLawService.storeCookie(this.name, this.expiration);
        this.seen = true;
        this.isSeen.emit(true);
    }
    /**
     * @return {?}
     */
    dismiss() {
        this.cookieLawComponent.dismiss();
    }
}
CookieLawContainerComponent.decorators = [
    { type: Component, args: [{
                selector: 'cookie-law',
                template: `
    <cookie-law-component
      *ngIf="!seen"
      [awsomeCloseIcon]="awsomeCloseIcon"
      [learnMore]="learnMore"
      [target]="target"
      [position]="position"
      (isSeen)="hasBeenDismissed()"
    >
      <ng-content></ng-content>
    </cookie-law-component>
  `
            },] },
];
/** @nocollapse */
CookieLawContainerComponent.ctorParameters = () => [
    { type: Angular2CookieLawService, },
];
CookieLawContainerComponent.propDecorators = {
    "seen": [{ type: HostBinding, args: ['attr.seen',] },],
    "cookieLawComponent": [{ type: ViewChild, args: [CookieLawComponent,] },],
    "name": [{ type: Input },],
    "learnMore": [{ type: Input },],
    "target": [{ type: Input },],
    "position": [{ type: Input },],
    "expiration": [{ type: Input },],
    "awsomeCloseIcon": [{ type: Input },],
    "isSeen": [{ type: Output },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class CookieLawModule {
    /**
     * @param {?} parentModule
     */
    constructor(parentModule) {
        if (parentModule) {
            throw new Error('CookieLawModule is already loaded. Import it in the AppModule only');
        }
    }
}
CookieLawModule.decorators = [
    { type: NgModule, args: [{
                imports: [CommonModule],
                declarations: [CookieLawComponent, CookieLawContainerComponent],
                exports: [CookieLawContainerComponent]
            },] },
];
/** @nocollapse */
CookieLawModule.ctorParameters = () => [
    { type: CookieLawModule, decorators: [{ type: Optional }, { type: SkipSelf },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { Angular2CookieLawService, CookieLawContainerComponent, CookieLawComponent, CookieLawModule, translateInOut as ɵa };
